modules/gcloud/bigquery.go

```diff
@@ -27,18 +27,24 @@ func RunBigQueryContainer(ctx context.Context, opts ...testcontainers.ContainerC
 
 	req.Cmd = []string{"--project", settings.ProjectID}
 
+	for _, opt := range opts {
+		if err := opt.Customize(&req); err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
Mmm I think the [applyOptions](https://github.com/testcontainers/testcontainers-go/pull/2523/files#diff-8faaed968761085a8ca6d01c0502d2c6159fabf32bec52d69cd76a86e56ecb87R23) call will already do this. Please see https://github.com/testcontainers/testcontainers-go/pull/2523/files#diff-658ed21fe179e3c4810748c731c20b75757c269dd8ce30f0bf98ce31f2addd24R91-R103

Reply:
Let me test this locally, from this branch. Will come back to you later

Reply:
Indeed, with the addition of the YAML data file, now it's possible to have different settings contributing to the CMD. So it's fine if we append.

My only concern is now related to the usage of the gcloud options across all the emulators 🤔 Do you know if all of them would support passing this YAML file? Else, anybody could add this functional option to e.g. Firestore or Pubsub, and expect something else. I'm not an expert in those services, so you probably have more insights.

Reply:
Good point, I myself have not used the other emulators but would assume this data yaml functionality would only apply for the bigquery emulator. I'm currently not sure of a great way to prevent this in the other gcloud emulators. I think the data.yaml option should be ignored though due to how that CMD is being set.

Reply:
I'm only processing the yaml data file in the bigQuery container. Until we have a better abstraction, this is fine I think